lib/repl.js

```diff
@@ -104,6 +108,20 @@ for (var n = 0; n < GLOBAL_OBJECT_PROPERTIES.length; n++) {
 const kBufferedCommandSymbol = Symbol('bufferedCommand');
 const kContextId = Symbol('contextId');
 
+let tmpListeners = [];
+let checkUncaught = false;
+let currentListeners = [];
```

Comment:
Why is this global state, and not per REPL instance?

Reply:
No REPL instance is allows to add any uncaught exception listeners. If they do, they'll be added to the `currentListeners` array and immediately be removed right after the code has been executed (evaluated). So the single listener is just right and it'll work for `n` REPL intances.

If each REPL instance would track this on their own, we'd add one event listener per instance to track this and that would trigger the event leak warning eventually.